src/test/stress/stress_test.go

```diff
@@ -0,0 +1,792 @@
+package stress
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	cryptorand "crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/shutter-network/encrypting-rpc-server/rpc"
+	sequencerBindings "github.com/shutter-network/gnosh-contracts/gnoshcontracts/sequencer"
+	shopContractBindings "github.com/shutter-network/shop-contracts/bindings"
+	"github.com/shutter-network/shutter/shlib/shcrypto"
+	"gotest.tools/assert"
+)
+
+func skipCI(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		t.Skip("Skipping testing in CI environment")
+	}
+}
+
+// the ethereum address of the key broadcast contract
+const KeyBroadcastContractAddress = "0x1FD85EfeC5FC18f2f688f82489468222dfC36d6D"
+
+// the ethereum address of the sequencer contract
+const SequencerContractAddress = "0xd073BD5A717Dce1832890f2Fdd9F4fBC4555e41A"
+
+// the ethereum address of the keyper set manager contract
+const KeyperSetManagerContractAddress = "0x7Fbc29C682f59f809583bFEE0fc50F1e4eb77774"
+
+const RpcUrl = "https://rpc.chiado.gnosis.gateway.fm"
+
+const KeyperSetChangeLookAhead = 2
+
+func createSetup(fundNewAccount bool) (StressSetup, error) {
+	setup := new(StressSetup)
+	client, err := ethclient.Dial(RpcUrl)
+	if err != nil {
+		return *setup, fmt.Errorf("could not create client %v", err)
+	}
+
+	setup.Client = client
+
+	chainID, err := client.NetworkID(context.Background())
+	if err != nil {
+		return *setup, fmt.Errorf("could not query chainId %v", err)
+	}
+	setup.ChainID = chainID
+
+	signerForChain := types.LatestSignerForChainID(chainID)
+	setup.SignerForChain = signerForChain
+
+	submitKeyHex := os.Getenv("STRESS_TEST_PK")
+	if len(submitKeyHex) < 64 {
```

Comment:
ooc, why is this condition checking if the ENV variable exists?

Reply:
go docs
```go
func os.Getenv(key string) string
/*
Getenv retrieves the value of the environment variable named by the key. 
It returns the value, which will be empty if the variable is not present. 
To distinguish between an empty value and an unset value, use LookupEnv.
*/
```
seemed to be sufficient for this use case.